src/passes/Precompute.cpp

```diff
@@ -829,7 +829,8 @@ struct Precompute
       return true;
     }
     // We can emit a StringConst for a string constant.
-    if (type.isString()) {
+    if (type.isRef() && type.getHeapType().isMaybeShared(HeapType::string)) {
+      assert(!type.getHeapType().isShared() && "TODO: shared string support");
```

Comment:
Why assert here? Why should we not be returning true?

Reply:
This still returns true for unshared strings, just like it did before. The new assertion prevents us from falling through and returning `false`. We don't want to try precomputing shared strings because they are generally not supported, tested, or fuzzed, so reporting that we can correctly precompute them is almost certainly incorrect at the moment, and in fact we should never be attempting to precompute them at all.

Reply:
What needs to be done for shared strings? They are immutable so modifications are not an issue. What kind of bug are you thinking of when you say it would almost certainly be incorrect to allow shared ones here?

Reply:
I expect we would run into some other assertion elsewhere or one of the bugs where the code hasn't been updated to account for shared types yet. I know for a fact that if we let the fuzzer get its hands on a test for shared strings, it will start failing assertions.

There may not be any fundamental problem with supporting shared strings, but it's just not something I would expect to work yet.

Reply:
Makes sense, but wouldn't it be best then to remove this assertion and let breakage happen elsewhere, if there is any? Otherwise, if we leave it, it's just an assertion we need to remove later. Or are you thinking this would be a canary to verify that we never run on shared strings at all? If the latter then maybe add a comment,
```
// There is no problem to allow shared strings to run in this pass, but error on it
// because other places are not yet ready, and this happens to be a convenient
// place to check.
```

Reply:
It seems better to fail early if we suspect we're going to fail later anyway, but you're right that this is just another assertion to remove later. I'll remove it since it's probably not pulling its weight.